daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -693,21 +696,374 @@ func TestPostTransaction(t *testing.T) {
+				CreatedAppID: futureAppID,
+			})
+			appCallTxn.ApprovalProgram = scenario.Program
+
+			txntest.Group(&payTxn, &appCallTxn)
+
+			stxns := []transactions.SignedTxn{
+				payTxn.Txn().Sign(sender.Secrets()),
+				appCallTxn.Txn().Sign(sender.Secrets()),
+			}
+
+			// build request body
+			var body io.Reader
+			var bodyBytes []byte
+			for _, stxn := range stxns {
+				bodyBytes = append(bodyBytes, protocol.Encode(&stxn)...)
+			}
+
+			msgpackFormat := model.SimulateTransactionParamsFormatMsgpack
+			jsonFormat := model.SimulateTransactionParamsFormatJson
+			responseFormats := []struct {
+				name   string
+				params model.SimulateTransactionParams
+				handle codec.Handle
+			}{
+				{
+					name: "msgpack",
+					params: model.SimulateTransactionParams{
+						Format: &msgpackFormat,
+					},
+					handle: protocol.CodecHandle,
+				},
+				{
+					name: "json",
+					params: model.SimulateTransactionParams{
+						Format: &jsonFormat,
+					},
+					handle: protocol.JSONStrictHandle,
+				},
+				{
+					name: "default",
+					params: model.SimulateTransactionParams{
+						Format: nil, // should default to JSON
+					},
+					handle: protocol.JSONStrictHandle,
+				},
+			}
+
+			for _, responseFormat := range responseFormats {
+				t.Run(string(responseFormat.name), func(t *testing.T) { //nolint:paralleltest // Uses shared testing env
+					body = bytes.NewReader(bodyBytes)
+					req := httptest.NewRequest(http.MethodPost, "/", body)
+					rec := httptest.NewRecorder()
+
+					e := echo.New()
+					c := e.NewContext(req, rec)
+
+					// simulate transaction
+					err := handler.SimulateTransaction(c, responseFormat.params)
+					require.NoError(t, err)
+					require.Equal(t, 200, rec.Code, rec.Body.String())
+
+					// decode actual response
+					var actualBody model.SimulationResponse
+					decoder := codec.NewDecoderBytes(rec.Body.Bytes(), responseFormat.handle)
+					err = decoder.Decode(&actualBody)
+					require.NoError(t, err)
+
+					var expectedFailedAt *[]uint64
+					if len(scenario.FailedAt) != 0 {
+						clone := make([]uint64, len(scenario.FailedAt))
+						copy(clone, scenario.FailedAt)
+						clone[0]++
+						expectedFailedAt = &clone
+					}
+					expectedBody := model.SimulationResponse{
+						Version: 1,
+						TxnGroups: []model.SimulationTransactionGroupResult{
+							{
+								FailedAt: expectedFailedAt,
+								TxnResults: []model.SimulationTransactionResult{
+									{
+										TxnResult: makePendingTxnResponse(t, transactions.SignedTxnWithAD{
+											SignedTxn: stxns[0],
+											// expect no ApplyData info
+										}, responseFormat.handle),
+									},
+									{
+										TxnResult: makePendingTxnResponse(t, transactions.SignedTxnWithAD{
+											SignedTxn: stxns[1],
+											ApplyData: scenario.ExpectedSimulationAD,
+										}, responseFormat.handle),
+									},
+								},
+							},
+						},
+						WouldSucceed: scenario.Outcome == mocktracer.ApprovalOutcome,
```

Comment:
shouldn't there be something about missing-signatures here?

Reply:
Not in this case, since all the transactions are properly signed (see line 919)